scipy/ndimage/src/_rank_filter_1d.cpp

```diff
@@ -157,11 +157,7 @@ void _rank_filter(T *in_arr, int rank, int arr_len, int win_len, T *out_arr,
   int lim2 = arr_len - lim;
```

Comment:
I think the other issue might be that this isn't validated to not be negative. Digging into it currently.

Reply:
Adding `if (lim2 < 0) return;` on the next line seems to stop the heap corruption. Do you see the same thing?